pkg/webui/console/views/device-add/manual/form/form.js

```diff
@@ -201,55 +231,141 @@ const ManualForm = props => {
       )
 
       return setActivationMode(mode)
     },
-    [validationContext],
+    [defaultNsSettings, macSettings, validationContext],
   )
 
-  const [defaultNsSettings, setDefaultNsSettings] = React.useState(true)
-  const handleDefaultNsSettings = React.useCallback(
-    checked => {
+  const [deviceClass, setDeviceClass] = React.useState(
+    initialValues._activation_mode === ACTIVATION_MODES.OTAA ? DEVICE_CLASSES.CLASS_A : undefined,
+  )
+  const handleDeviceClassChange = React.useCallback(
+    devClass => {
       const { setValues, values } = formRef.current
 
+      setDeviceClass(devClass)
       setValues(
         validationSchema.cast(
           {
             ...defaultValues,
             ...values,
-            mac_settings: defaultValues.mac_settings,
+            mac_settings: macSettings,
           },
           { context: validationContext },
         ),
       )
-
-      return setDefaultNsSettings(checked)
     },
-    [validationContext],
-  )
-
-  const [deviceClass, setDeviceClass] = React.useState(
-    initialValues._activation_mode === ACTIVATION_MODES.OTAA ? DEVICE_CLASSES.CLASS_A : undefined,
+    [macSettings, validationContext],
   )
-  const handleDeviceClassChange = React.useCallback(devClass => {
-    setDeviceClass(devClass)
-  }, [])
 
   const [freqPlan, setFreqPlan] = React.useState()
+  const freqPlanRef = React.useRef(freqPlan)
   const handleFreqPlanChange = React.useCallback(band => {
     setFreqPlan(band.value)
   }, [])
 
+  const [phyVersion, setPhyVersion] = React.useState()
+  const phyVersionRef = React.useRef(phyVersion)
+  const handlePhyVersionChange = React.useCallback(version => {
+    setPhyVersion(version)
+  }, [])
+
+  React.useEffect(() => {
+    const updateMacSettings = async () => {
+      try {
+        const settings = await getDefaultMacSettings(freqPlan, phyVersion)
+        setMacSettings(settings)
+      } catch (err) {
+        toast({
+          type: toast.types.ERROR,
+          message: m.macSettingsError,
+          messageValues: freqPlan,
+        })
+      }
+    }
+    const resetMacSettings = () => {
+      setMacSettings({})
+    }
+
+    if (freqPlan && phyVersion) {
+      if (freqPlanRef.current !== freqPlan || phyVersionRef.current !== phyVersion) {
+        freqPlanRef.current = freqPlan
+        phyVersionRef.current = phyVersion
+
+        updateMacSettings()
+      }
+    } else {
+      resetMacSettings()
+    }
+  }, [freqPlan, getDefaultMacSettings, phyVersion])
+
+  React.useEffect(() => {
+    if (formRef.current) {
+      if (!isEqual(macSettings, macSettingsRef.current)) {
+        const { setValues, values } = formRef.current
+
+        setValues(
+          validationSchema.cast(
+            { ...values, mac_settings: { ...values.mac_settings, ...macSettings } },
+            { context: validationContext },
+          ),
+        )
+
+        macSettingsRef.current = macSettings
+      }
+    }
+  }, [macSettings, validationContext])
```

Comment:
I'm not quite getting this approach. What's the reason for using a side effect to modify the form as opposed to using a hook such as `onMacSettingsChange()` to be passed to the `<AdvancedSettingsSection />` ? 

Same for the logic above to handle frequency plan updates.